### PR TITLE
Add lastResourceOptimisationInterfaceClassName and commitMarkableResourceJNDINames to JTAEnvironmentBean for better integration with agroal

### PR DESCRIPTION
**Pull request by graben**
Enhancement based on feedback from agroal/agroal#69. Implementation analog to WildFly [JTAEnvironmentBeanService.java#L61](https://github.com/wildfly/wildfly/blob/29.0.0.Final/transactions/src/main/java/org/jboss/as/txn/service/JTAEnvironmentBeanService.java#L61)

### CODE REVIEW

File: narayana-spring-boot-core/src/main/java/me/snowdrop/boot/narayana/core/properties/NarayanaPropertiesInitializer.java [hunk 1]
```diff
@@ -104,6 +106,15 @@ private void setXAResourceRecordWrappingPlugin(String xaResourceRecordWrappingPl
         getPopulator(JTAEnvironmentBean.class).setXaResourceRecordWrappingPluginClassName(xaResourceRecordWrappingPlugin);
     }
 
+    private void setLastResourceOptimisationInterface(String lastResourceOptimisationInterface) {
+        getPopulator(JTAEnvironmentBean.class).setLastResourceOptimisationInterfaceClassName(lastResourceOptimisationInterface);
+    }
+
+    private void setCommitMarkableResourceJNDINames(List<String> commitMarkableResourceJNDINames) {
+        getPopulator(JTAEnvironmentBean.class).setPerformImmediateCleanupOfCommitMarkableResourceBranches(true);
```
Comment (barreiro) [inline narayana-spring-boot-core/src/main/java/me/snowdrop/boot/narayana/core/properties/NarayanaPropertiesInitializer.java:114]: Agroal will not support this. See [AG-214](https://issues.redhat.com/projects/AG/issues/AG-214) for further details. I also don't see an advantage on immediate cleanup, the recovery manager can do it in batch which ends up being more efficient.

File: narayana-spring-boot-core/src/main/java/me/snowdrop/boot/narayana/core/properties/NarayanaPropertiesInitializer.java [hunk 2]
```diff
@@ -104,6 +106,15 @@ private void setXAResourceRecordWrappingPlugin(String xaResourceRecordWrappingPl
         getPopulator(JTAEnvironmentBean.class).setXaResourceRecordWrappingPluginClassName(xaResourceRecordWrappingPlugin);
     }
 
+    private void setLastResourceOptimisationInterface(String lastResourceOptimisationInterface) {
+        getPopulator(JTAEnvironmentBean.class).setLastResourceOptimisationInterfaceClassName(lastResourceOptimisationInterface);
```
Comment (barreiro) [inline narayana-spring-boot-core/src/main/java/me/snowdrop/boot/narayana/core/properties/NarayanaPropertiesInitializer.java:110]: :+1:

**Review by barreiro — CHANGES_REQUESTED** [review]

**Comment by graben** [comment]
@Sgitario : Could you plz review and merge

**Comment by Sgitario** [comment]
@barreiro can you have a final review on this pull request? Thanks!

**Comment by graben** [comment]
@Sgitario : Ready to merge and probably do a first release of 3.0?

**Comment by Sgitario** [comment]
Merged.
About releasing, cc @jacobdotcosta

**Comment by graben** [comment]
@Sgitario : Well, last time @jacobdotcosta was asked for a release, nothing happens. Anyone else able to do releases?

**Comment by Sgitario** [comment]
> @Sgitario : Well, last time @jacobdotcosta was asked for a release, nothing happens. Anyone else able to do releases?

No that I'm aware of. @jacobdotcosta is on PTO until 28 Aug.

**Pull request merged**